src/Datadog.Trace/Tracer.cs

```diff
@@ -415,6 +417,91 @@ internal void StartDiagnosticObserversInternal()
             }
         }
 
+        internal void WriteDiagnosticLog()
+        {
+            try
+            {
+                var stringWriter = new StringWriter();
+
+                using (var writer = new JsonTextWriter(stringWriter))
+                {
+                    writer.WriteStartObject();
+
+                    writer.WritePropertyName("date");
+                    writer.WriteValue(DateTime.Now);
+
+                    writer.WritePropertyName("os_name");
+#if NETSTANDARD
+                    writer.WriteValue(System.Runtime.InteropServices.RuntimeInformation.OSDescription);
+#else
+                    writer.WriteValue(Environment.OSVersion.ToString());
+#endif
+
+                    writer.WritePropertyName("version");
+                    writer.WriteValue(typeof(Tracer).Assembly.GetName().Version.ToString());
+
+                    writer.WritePropertyName("lang");
+                    writer.WriteValue(".net");
+
+                    writer.WritePropertyName("lang_version");
+#if NETSTANDARD
+                    writer.WriteValue(System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription);
+#else
+                    writer.WriteValue(Environment.Version.ToString());
+#endif
```

Comment:
We have a `FrameworkDescription` helper class which takes care of some problems with using `Environment.Version`.

Reply:
Nice!